src/subscriber.ts

```diff
@@ -172,15 +172,22 @@ export class Subscriber {
       const isHeartbeatExpected = await this._isHeadAfterHeartbeatBlockThreshold(header)
       const sessionIndex = await this.api.query.session.currentIndex()
 
-     this.validators.forEach(async account => {
-        if( isHeartbeatExpected && ! await this._hasValidatorAuthoredBlocks(account,sessionIndex) && ! await this._hasValidatorSentHeartbeat(account,sessionIndex) ){
-          this.logger.info(`Target ${account.name} has either not authored any block or sent any heartbeat yet`);
-          this.promClient.setStateValidatorOfflineSessionReports(account.name)
+      this.validators.forEach(async account => {
+
+        if(isHeartbeatExpected) {
+          if ( await this._hasValidatorAuthoredBlocks(account,sessionIndex) || await this._hasValidatorSentHeartbeat(account,sessionIndex) ) {
+            this.promClient.resetStatusValidatorOffline(account.name);
+          }
+          else {
+            this.logger.info(`Target ${account.name} has either not authored any block or sent any heartbeat yet`);
+            this.promClient.setStatusValidatorOffline(account.name);
+          }
         }
-        else{
-            this.promClient.resetStateValidatorOfflineSessionReports(account.name)
+        else if ( this.promClient.isValidatorStatusOffline && ( await this._hasValidatorAuthoredBlocks(account,sessionIndex) || await this._hasValidatorSentHeartbeat(account,sessionIndex) )) {
```

Comment:
Shouldn't we call here `this.promClient.isValidatorStatusOffline(account.name)`?

Reply:
Also maybe we could separate the two if conditions as we have above, something like:
```
else if(this.promClient.isValidatorStatusOffline(account.name)){
  if(await this._hasValidatorAuthoredBlocks(account,sessionIndex) || await this._hasValidatorSentHeartbeat(account,sessionIndex)){
    .....
  }
} 
```
wdyt? Given that `await this._hasValidatorAuthoredBlocks(account,sessionIndex) || await this._hasValidatorSentHeartbeat(account,sessionIndex)` maybe we could refactor it in a separate method?

Reply:
totally agree with you, now it's even cleaner In my opinion